custom_components/frigate/media_source.py

```diff
@@ -736,18 +695,30 @@ async def async_browse_media(
             )
 
         if isinstance(identifier, RecordingIdentifier):
-            path = identifier.get_integration_proxy_path()
             try:
-                recordings_folder = await self._get_client(identifier).async_get_path(
-                    path
+                if not identifier.camera:
+                    config = await self._get_client(identifier).async_get_config()
+                    return self._get_camera_recording_identifiers(identifier, config)
+
+                if not identifier.year_month_day:
+                    days = cast(
+                        list[dict[str, Any]],
+                        await self._get_client(identifier).async_get_recordings_summary(
+                            camera=identifier.camera
+                        ),
+                    )
+                    return self._get_recording_folders(identifier, days)
+
+                hours = cast(
```

Comment:
Nit: Both the hours and days variables are set to the same content (I think?), can probably make this code slightly cleaner by only making the API call once, then calling the different methods after that.